cmake/CorrosionGenerator.cmake

```diff
@@ -0,0 +1,519 @@
+function(_cargo_metadata out manifest)
+    get_property(
+        RUSTC_EXECUTABLE
+        TARGET Rust::Rustc PROPERTY IMPORTED_LOCATION
+    )
+    get_property(
+        CARGO_EXECUTABLE
+        TARGET Rust::Cargo PROPERTY IMPORTED_LOCATION
+    )
+    execute_process(
+        COMMAND
+            ${CMAKE_COMMAND} -E env
+                CARGO_BUILD_RUSTC=${RUSTC_EXECUTABLE}
+                ${CARGO_EXECUTABLE}
+                    metadata
+                        --manifest-path ${manifest}
+                        --format-version 1
+        OUTPUT_VARIABLE json
+        COMMAND_ERROR_IS_FATAL ANY
+    )
+
+    set(${out} ${json} PARENT_SCOPE)
+endfunction(_cargo_metadata)
+
+function(_generator_print_root out manifest)
```

Comment:
The original rust command actually printed to stdout, but here there is no printing, so I think it would make sense to adjust the function name

Reply:
done